modules/collect/src/main/java/com/opengamma/strata/collect/Messages.java

```diff
@@ -146,52 +148,34 @@ public static String format(String messageTemplate, Object... args) {
       return formatWithAttributes("", args);
     }
     if (args == null) {
-      return formatWithAttributes(messageTemplate, new Object[0]);
+      return formatWithAttributes(messageTemplate);
     }
 
-    // try to make builder big enough for the message and the args
-    StringBuilder outputStringBuilder = new StringBuilder(messageTemplate.length() + args.length * 20);
-    Map<String, String> attributesMap = new HashMap<>();
-
-    String pattern = "\\{(\\w*)\\}"; //This will match both {}, and {anything}
-    Pattern regexPattern = Pattern.compile(pattern);
-    Matcher matcher = regexPattern.matcher(messageTemplate);
+    Map<String, String> attributes = new HashMap<>();
+    Matcher matcher = REGEX_PATTERN.matcher(messageTemplate);
     int groupIndex = 0;
-    int lastAddedStringEndIndex = 0;
+
+    StringBuffer outputMessageBuffer = new StringBuffer();
     int matchesCount = 0;
     while (matcher.find()) {
       matchesCount++;
-    }
-    if (matchesCount > args.length) {
-      throw new IllegalArgumentException(
-          Messages.format(
-              "You have included {} placeholders, however only provided {} arguments.",
-              matchesCount,
-              args.length));
-    }
-    matcher.reset(); //Since we have already called .find(), and that state needs to be reset
-    while (matcher.find()) {
-      String attributeName = matcher.group(1); //Extract the attribute name
-      int startIndex = matcher.start();
-      int endIndex = matcher.end();
 
+      if (matchesCount > args.length) {
+        throw new IllegalArgumentException(
```

Comment:
Don't throw an exception. Just ignore the extra placeholders so they appear in the output as `{}` or `{foo}`.